synapse/federation/transport/server.py

```diff
@@ -824,7 +824,7 @@ def __init__(self, handler, authenticator, ratelimiter, server_name, allow_acces
         if not self.allow_access:
             raise FederationDeniedError(origin)
 
-        limit = int(content.get("limit", 100))
+        limit = int(content.get("limit", 100))  # type: Optional[int]
```

Comment:
Will `limit` ever be `None`? `int(None)` raises a `TypeError`.

Reply:
Limit shouldn't be none here, but later we'll set it to `None` if its zero